source/fundamentals/data-formats/custom-types.txt

```diff
@@ -1,348 +1,342 @@
-  ``~bson.decimal128.Decimal128`` value needs to be decoded as ``Decimal``
-  and which needs to be decoded as ``DecimalInt``. This example only considers
-  the situation where a user wants to *encode* documents containing either
-  of these types.
-
-After creating a new codec options object and using it to get a collection
-object, we can seamlessly encode instances of ``DecimalInt``:
+To encode an instance of the ``DecimalInt`` class, you must define a type codec for
+the class. This type codec must inherit from the parent class's codec:
 
 .. code-block:: python
 
-  >>> type_registry = TypeRegistry([decimal_codec, decimalint_codec])
-  >>> codec_options = CodecOptions(type_registry=type_registry)
-  >>> collection = db.get_collection("test", codec_options=codec_options)
-  >>> collection.drop()
-  >>> collection.insert_one({"num": DecimalInt("45.321")})
-  InsertOneResult(ObjectId('...'), acknowledged=True)
-  >>> mydoc = collection.find_one()
-  >>> pprint.pprint(mydoc)
-  {'_id': ObjectId('...'), 'num': Decimal('45.321')}
-
-Note that the ``transform_bson`` method of the base codec class results in
-these values being decoded as ``Decimal`` (and not ``DecimalInt``).
-
-
-.. _decoding-binary-types:
-
-Decoding ``~bson.binary.Binary`` Types
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-The decoding treatment of ``~bson.binary.Binary`` types having
-``subtype = 0`` by the ``bson`` module varies slightly depending on the
-version of the Python runtime in use. This must be taken into account while
-writing a ``TypeDecoder`` that modifies how this datatype is decoded.
-
-On Python 3.x, ``~bson.binary.Binary`` data (``subtype = 0``) is decoded
-as a ``bytes`` instance:
-
-.. code-block:: pycon
+   class DecimalIntCodec(DecimalCodec):
+       @property
+       def python_type(self):
+           # The Python type acted upon by this type codec
+           return DecimalInt
 
-    >>> # On Python 3.x.
-    >>> from bson.binary import Binary
-    >>> newcoll = db.get_collection("new")
-    >>> newcoll.insert_one({"_id": 1, "data": Binary(b"123", subtype=0)})
-    >>> doc = newcoll.find_one()
-    >>> type(doc["data"])
-    bytes
+You can then add the sublcass's type codec to the type registry and encode instances
+of the custom type:
 
+.. io-code-block::
+   :copyable: true
 
-On Python 2.7.x, the same data is decoded as a ``~bson.binary.Binary``
-instance:
+   .. input::
+      :language: python
 
-.. code-block:: pycon
+      import pprint
+      from bson.codec_options import CodecOptions
 
-    >>> # On Python 2.7.x
-    >>> newcoll = db.get_collection("new")
-    >>> doc = newcoll.find_one()
-    >>> type(doc["data"])
-    bson.binary.Binary
+      decimal_int_codec = DecimalIntCodec()
+      type_registry = TypeRegistry([decimal_codec, decimal_int_codec])
+      codec_options = CodecOptions(type_registry=type_registry)
 
+      collection = db.get_collection("test", codec_options=codec_options)
+      collection.drop()
```

Comment:
S: Remove collection.drop() since we aren't expecting them to run each of these examples iteratively. I don't think we want to make it seem like you have to drop the collection any time you do this

```suggestion
      collection = db.get_collection("test", codec_options=codec_options)

```

Reply:
yeah agreed